internal/docstore/drivertest/drivertest.go

```diff
@@ -53,6 +54,12 @@ type Harness interface {
 	// that the revision field should be drivertest.AlternateRevisionField.
 	MakeAlternateRevisionFieldCollection(context.Context) (driver.Collection, error)
 
+	// BeforeDoTypes should return a list of values whose types are valid for the as
+	// function given to BeforeDo. For example, if the provider converts Get actions
```

Comment:
It should be clearer from this whether `As` should work for all of the types, or just one of them. The test should probably support both, as there are cases where a single `As` will support multiple types (not one or the other, but both).

Reply:
I try BeforeDo on all six action types, and each action type must support at least one of the BeforeDo types.

I thought about making this more fine-grained, but it's tricky. A Get may have one type, a Put another, and an Update two others. It's going to get messy to specify all that. I figured this was a good start.

How about I put a TODO to make it more action-specific?

Reply:
I added the TODO.